### ISSUE TITLE
incorrect limit (radius of convergence)

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#4044]
Imported from SourceForge on 2024-07-09 10:44:59
Created by **[claudiusmaximus](https://sourceforge.net/u/claudiusmaximus/)** on 2021-04-07 16:16:49
Original: https://sourceforge.net/p/maxima/bugs/3767

---

Hi, I am trying to work out the radius of convergence of some Taylor series related to Newton's method applied to points in the Mandelbrot set.  I found Maxima calculates a limit incorrectly (it should be 2, not inf):

~~~
f(c, z) := z^2 + c;
h(c) := c^2 + c; /* f(c, f(c, 0)); */
hh(c) := 2 * c + 1; /* diff(h(c), c); */
N(z) := z - h(z) / hh(z);
M(z) := subst(c = c + z, N(c)) - c;
M(z);
for ct in solve(h(c) = 0) do
(
  print(ct),
  print(M(z)),
  a(n) := limit(diff(subst(ct, M(z)), z, n) / factorial(n), z, 0),
  print(a(0), a(1), a(2), a(3), a(4), a(5)),
  print(1 / abs(a(11)/a(10)), 1 / abs(a(101)/a(100)), 1 / abs(a(1001)/a(1000))),
  R : 1 / limit(abs(a(n+1)/a(n)), n, infinity),
  print(R) /* should give R = 1/2; gives R = 1/inf */
);
~~~

Build information:
~~~
Maxima version: "5.44.0"
Maxima build date: "2020-08-29 21:03:14"
Host type: "x86_64-pc-linux-gnu"
Lisp implementation type: "GNU Common Lisp (GCL)"
Lisp implementation version: "GCL 2.6.12"
User dir: "/home/claude/.maxima"
Temp dir: "/tmp"
Object dir: "/home/claude/.maxima/binary/5_44_0/gcl/GCL_2_6_12"
Frontend: false
~~~

This is the version from Debian Bullseye (current testing), though I tried the Buster version and it had the same issue:
~~~
$ apt-cache policy maxima
maxima:
  Installed: 5.44.0-2
  Candidate: 5.44.0-2
  Version table:
 *** 5.44.0-2 990
        990 http://ftp.uk.debian.org/debian bullseye/main amd64 Packages
        990 http://ftp.uk.debian.org/debian testing/main amd64 Packages
        500 http://ftp.uk.debian.org/debian unstable/main amd64 Packages
        100 /var/lib/dpkg/status
     5.42.1-1 500
        500 http://ftp.uk.debian.org/debian buster/main amd64 Packages
~~~

Let me know if you need further information.

Claude

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-09 10:45:00
Created by **[claudiusmaximus](https://sourceforge.net/u/claudiusmaximus/)** on 2021-04-08 11:11:28
Original: https://sourceforge.net/p/maxima/bugs/3767/#7a91

---

Same issue with current git from sf:

~~~
Maxima version: "branch_5_44_base_372_g042a6e8a7"
Maxima build date: "2021-04-08 12:08:35"
Host type: "x86_64-pc-linux-gnu"
Lisp implementation type: "GNU Common Lisp (GCL)"
Lisp implementation version: "GCL 2.6.12"
User dir: "/home/claude/.maxima"
Temp dir: "/tmp"
Object dir: "/home/claude/.maxima/binary/branch_5_44_base_372_g042a6e8a7/gcl/GCL_2_6_12"
Frontend: false
~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-09 10:45:04
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2021-04-15 12:16:22
Original: https://sourceforge.net/p/maxima/bugs/3767/#20e7

---

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-09 10:45:07
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2021-04-15 12:16:22
Original: https://sourceforge.net/p/maxima/bugs/3767/#2ca1

---

The function limit is called on things like

~~~
(abs((n+1)!)*abs(limit('diff((z-(z^2+z)/(2*z+1)),z,n),z,0)))/(abs(n!)*abs(limit('diff((z-(z^2+z)/(2*z+1)),z,n+1),z,0)))
~~~

Unfortunately,  Maxima knows nothing about the derivative of any function with a symbolic order. So limit should return a nounform for this case.  But it doesn't. 

A simplified version of this bug is

~~~
(%i16)	limit(diff(exp(n * x), x,n)/n,n,inf);
(%o16)	0
~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-09 10:45:11
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2021-05-26 13:42:00
Original: https://sourceforge.net/p/maxima/bugs/3767/#486b

---

- **status**: open --> closed
- **assigned_to**: Dan Gildea

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-09 10:45:15
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2021-05-26 13:42:01
Original: https://sourceforge.net/p/maxima/bugs/3767/#d42e

---

Fixed by [5a4f092d69a12ae66ad60de1243ae46c489fabc5]
